core/headerchain.go

```diff
@@ -97,12 +97,7 @@ func NewHeaderChain(chainDb ethdb.Database, config *params.ChainConfig, getValid
 
 	hc.genesisHeader = hc.GetHeaderByNumber(0)
 	if hc.genesisHeader == nil {
-		genesisBlock, err := WriteDefaultGenesisBlock(chainDb)
-		if err != nil {
-			return nil, err
-		}
-		log.Warn("Wrote default Ethereum genesis block")
-		hc.genesisHeader = genesisBlock.Header()
+		return nil, ErrNoGenesis
```

Comment:
LightEthereum writes the genesis block using `core.SetupGenesisBlock` before creating LightChain and writing the default block was already gone in BlockChain. I just removed it here because this case can't really happen (and couldn't happen previously).